### PR TITLE
LP-2624 webinar description page

### PR DESCRIPTION
**Pull request by aliassad**
[LP-2624](https://philanthropyu.atlassian.net/browse/LP-2624)
[adg-theme-PR](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/231)

Note: 
Integrate webinar description design

### CODE REVIEW

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1576/workflows/workflows_id/jobs/1576/artifacts)**. Coverage !!! :scream:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/views.py (0.0%): Missing lines 23
-------------
Total:   1 line
Missing: 1 line
Coverage: 0%
-------------
```

File: openedx/adg/lms/webinars/views.py [hunk 1]
```diff
@@ -16,8 +16,11 @@
 
 
 def webinar_description_page_view(request, pk):
-    # TODO This is a temp view and will be properly functional in LP-2623
-    return render(request, 'adg/lms/webinar/description_page.html', context={'pk': pk})
+    """
+       A view to get description about a specific webinar.
```
Comment (asadali145) [inline openedx/adg/lms/webinars/views.py:20]: Remove extra indent in docstring.

File: openedx/adg/lms/webinars/views.py [hunk 2]
```diff
@@ -16,8 +16,11 @@
...
+    """
+
```
Comment (asadali145) [inline openedx/adg/lms/webinars/views.py:22]: No need for this blank line as we have used function based view.

Comment (aliassad) [reply openedx/adg/lms/webinars/views.py:22]: right (Y)

File: openedx/adg/lms/webinars/views.py (hunk 1 above)
Comment (aliassad) [reply openedx/adg/lms/webinars/views.py:20]: Done

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1592/workflows/workflows_id/jobs/1592/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/tests/test_views.py (100%)
openedx/adg/lms/webinars/urls.py (100%)
openedx/adg/lms/webinars/views.py (100%)
-------------
Total:   7 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1593/workflows/workflows_id/jobs/1593/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/tests/test_views.py (100%)
openedx/adg/lms/webinars/urls.py (100%)
openedx/adg/lms/webinars/views.py (100%)
-------------
Total:   7 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1594/workflows/workflows_id/jobs/1594/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/tests/test_views.py (100%)
openedx/adg/lms/webinars/urls.py (100%)
openedx/adg/lms/webinars/views.py (100%)
-------------
Total:   7 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1595/workflows/workflows_id/jobs/1595/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/tests/test_views.py (100%)
openedx/adg/lms/webinars/urls.py (100%)
openedx/adg/lms/webinars/views.py (100%)
-------------
Total:   9 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1596/workflows/workflows_id/jobs/1596/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/tests/test_views.py (100%)
openedx/adg/lms/webinars/urls.py (100%)
openedx/adg/lms/webinars/views.py (100%)
-------------
Total:   7 lines
Missing: 0 lines
Coverage: 100%
-------------
```

File: openedx/adg/lms/webinars/tests/test_views.py [hunk 3]
```diff
@@ -32,6 +32,15 @@ def test_webinar_registration_view_object_does_not_exist(user_client):
     assert response.status_code == 404
 
 
+def test_webinar_description_view_invalid_pk(user_client):
+    """
+    Test webinar description with invalid pk
+    """
+    _, client = user_client
+    response = client.get(reverse('webinar_event', kwargs={'pk': 999}))
+    assert response.status_code == 404
```
Comment (asadali145) [inline openedx/adg/lms/webinars/tests/test_views.py:41]: Can we add a test for valid id as well?

File: openedx/adg/lms/webinars/views.py [hunk 4]
```diff
@@ -16,8 +16,11 @@
 
 
 def webinar_description_page_view(request, pk):
-    # TODO This is a temp view and will be properly functional in LP-2623
-    return render(request, 'adg/lms/webinar/description_page.html', context={'pk': pk})
+    """
+    A view to get description about a specific webinar.
+    """
+    return render(request, 'adg/lms/webinar/description_page.html',
+                  context={'webinar': get_object_or_404(Webinar, pk=pk)})
```
Comment (asadali145) [inline openedx/adg/lms/webinars/views.py:23]:
```suggestion
        return render(
        request, 'adg/lms/webinar/description_page.html', context={'webinar': get_object_or_404(Webinar, pk=pk)}
    )
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1598/workflows/workflows_id/jobs/1598/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/tests/test_views.py (100%)
openedx/adg/lms/webinars/urls.py (100%)
openedx/adg/lms/webinars/views.py (100%)
-------------
Total:   18 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Review by asadali145 — APPROVED** [review]

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1601/workflows/workflows_id/jobs/1601/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/tests/test_views.py (100%)
openedx/adg/lms/webinars/urls.py (100%)
openedx/adg/lms/webinars/views.py (100%)
-------------
Total:   17 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1602/workflows/workflows_id/jobs/1602/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/tests/test_views.py (100%)
openedx/adg/lms/webinars/urls.py (100%)
openedx/adg/lms/webinars/views.py (100%)
-------------
Total:   17 lines
Missing: 0 lines
Coverage: 100%
-------------
```

File: openedx/adg/lms/webinars/tests/test_views.py [hunk 5]
```diff
@@ -32,6 +36,34 @@ def test_webinar_registration_view_object_does_not_exist(user_client):
...
+
+
+@mock.patch('openedx.adg.lms.webinars.views.render')
+def test_webinar_description_view_valid_pk(mock_render):
```
Comment (zahid-arbisoft) [inline openedx/adg/lms/webinars/tests/test_views.py:49]: Please use the mocker fixture instead for mocking. See line # 80

File: openedx/adg/lms/webinars/views.py (hunk 4 above)
Comment (zahid-arbisoft) [inline openedx/adg/lms/webinars/views.py:23]: Can we use generic CBV for example [DetailView](https://docs.djangoproject.com/en/2.2/ref/class-based-views/generic-display/#detailview), we can update URL accordingly as well

**Review by zahid-arbisoft — CHANGES_REQUESTED** [review]

File: openedx/adg/lms/webinars/views.py (hunk 4 above)
Comment (ahmed-faraz46) [reply openedx/adg/lms/webinars/views.py:23]: +1

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1607/workflows/workflows_id/jobs/1607/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/tests/test_views.py (100%)
openedx/adg/lms/webinars/urls.py (100%)
openedx/adg/lms/webinars/views.py (100%)
-------------
Total:   18 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Review by zahid-arbisoft — APPROVED** [review]

**Pull request merged**